### PR TITLE
Native response handler

### PR DESCRIPTION
**Pull request by ramsessanchez**
Adds a native response handler for use in the LargeFileUpload task requests used in Java-Core for Kiota. 
Adds setResponseHandler function to the requestInformation class.

### CODE REVIEW

**Comment by baywet** [comment]
@ramsessanchez can you do a first pass removing the code smells reported above please before we proceed with a review?

**Review by andrueastman — APPROVED** [review]

File: components/http/okHttp/gradle/dependencies.gradle [hunk 1]
```diff
@@ -13,7 +13,7 @@ dependencies {
     implementation 'io.opentelemetry:opentelemetry-context:1.24.0'
     implementation 'io.opentelemetry:opentelemetry-semconv:1.24.0-alpha'
     implementation 'com.google.guava:guava:31.1-jre'
-    api 'com.squareup.okhttp3:okhttp:4.10.0'
+    implementation 'com.squareup.okhttp3:okhttp:4.10.0'
```
Comment (baywet) [inline components/http/okHttp/gradle/dependencies.gradle:16]: revert this change, we're exposing this type in the API surface of the request adapter

**Review by baywet — APPROVED** [review]
